tribits/core/package_arch/TribitsAddExecutable.cmake

```diff
@@ -652,7 +652,27 @@ function(tribits_add_executable EXE_NAME)
       message("-- ${EXE_NAME}:LINK_LIBS='${LINK_LIBS}'")
   endif()
 
-  target_link_libraries(${EXE_BINARY_NAME} PUBLIC ${LINK_LIBS})
+  #
+  # Link ${EXE_BINARY_NAME} to direct upstream libraries
+  #
+
+  target_link_libraries(${EXE_BINARY_NAME} PUBLIC ${${PACKAGE_NAME}_LIBRARIES})
+  foreach(depPkg IN LISTS ${PACKAGE_NAME}_LIB_ENABLED_DEPENDENCIES)
+    if (TARGET ${depPkg}::all_libs)
+      target_link_libraries(${EXE_BINARY_NAME} PUBLIC ${depPkg}::all_libs)
+    endif()
+  endforeach()
+  foreach(depPkg IN LISTS ${PACKAGE_NAME}_TEST_ENABLED_DEPENDENCIES)
+    if (TARGET ${depPkg}::all_libs)
+      target_link_libraries(${EXE_BINARY_NAME} PUBLIC ${depPkg}::all_libs)
+    endif()
```

Comment:
I see a lot of instances where you have the exact same logic for two different lists. You can combine both lists in `foreach(IN LISTS)`. For example:

```cmake
foreach(depPkg IN LISTS ${PACKAGE_NAME}_LIB_ENABLED_DEPENDENCIES ${PACKAGE_NAME}_TEST_ENABLED_DEPENDENCIES)
  if(TARGET ${depPkg}::all_libs)
    target_link_libraries(${EXE_BINARY_NAME} PUBLIC ${depPkg}::all_libs)
  endif()
endforeach()
```

Reply:
That is a good point.  Please flag other instances you see like this